openfaas/openfaas-cloud
# Issue 5 : Github Commit Status integration

This pull request implements the gitlab status integration 

This issue resolves: #5 

##### Change Overview
Openfaas-cloud will update status for repo based on Build and Deployment Status
Current status definition are being provided as: 
```
{
"state": "failure", // Current applicable values success / failure
"target_url": ":CloneURL",
"description": "Error String of actual Error",
"context": "BUILD". //  Current applicable values BUILD / DEPLOY
}
```

##### To Test: 
1. Gitlab App Should be created with read + write permission on `Commit Statuses`
![screen shot 2018-04-25 at 2 46 31 pm](https://user-images.githubusercontent.com/7374058/39228157-84114e98-4897-11e8-87b0-1be49cf70fff.png)

2. Get you gitlab APP Id
![screen shot 2018-04-25 at 2 47 24 pm](https://user-images.githubusercontent.com/7374058/39228182-bafda49c-4897-11e8-9473-37b3070fe729.jpg)

3. Add gitlab Id in your `github.yml`
```bash
environment:
    github_webhook_secret: "webhook secret"
    github_app_id: "app id"
```
3. Download Private Key
![screen shot 2018-04-25 at 2 49 51 pm](https://user-images.githubusercontent.com/7374058/39228225-f5f5aee6-4897-11e8-80ac-54735b3c97eb.png)

4. Create Docker Secret with Private Key
```bash
docker secret create private_key.pem <downloaded_pem_file_path>
```
5. Commit Something in your repo to get status published.  
or can be manually tested by
```bash
curl -X POST -H "Content-Type: application/json" -H "X_Github_Event: push" http://my.openfaas.cloud/function/gh-push -d '{"after": "<latest_commit_sha>","repository":{"name":"<foo>","full_name": "bar/foo","clone_url": "https://github.com/bar/foo.git","owner":{"login": "bar","email": "bar@email.com"}}, "installation":{"id": <your_ins_id>}}'
```

##### Example Repo: 
[regex_go](https://github.com/s8sg/regex_go) is integrated with openfaas 

<img width="962" alt="screen shot 2018-04-26 at 12 00 06 am" src="https://user-images.githubusercontent.com/7374058/39254383-ea1a2cf8-48e4-11e8-8f99-9113f30d035c.png">

With Application: [s8sgofcloud](https://github.com/apps/s8sgofcloud)
